### ISSUE TITLE
Increasing cmdheight on opening menu

### ISSUE THREAD

**Issue by sirex** [preservim/nerdtree#1333]
#### Self-Diagnosis
Before creating an issue, take some time to search these resources for an answer. It's possible that someone else has already seen and solved your issue.
- [old NERDTree issues](https://github.com/preservim/nerdtree/issues?q=is%3Aissue)
- NERDTree documentation - `:h NERDTree`
- [NERDTree Wiki](https://github.com/preservim/nerdtree/wiki)
- Other resources: <https://stackoverflow.com>, <https://vi.stackexchange.com>, etc.

#### Environment
- Operating System:  Manjaro 22.0.0 Sikaris
- Vim/Neovim version `:version`: NVIM v0.8.0
- NERDTree version, found on first line of quickhelp `?`: 6.10.16
- Are you using any of these NERDTree-dependent plugins?  
    - [ ] [Xuyuanp/nerdtree-git-plugin](https://github.com/Xuyuanp/nerdtree-git-plugin)
    - [ ] [ryanoasis/vim-devicons](https://github.com/ryanoasis/vim-devicons)
    - [ ] [tiagofumo/vim-nerdtree-syntax-highlight](https://github.com/tiagofumo/vim-nerdtree-syntax-highlight)
    - [ ] [scrooloose/nerdtree-project-plugin](https://github.com/scrooloose/nerdtree-project-plugin)
    - [ ] [PhilRunninger/nerdtree-buffer-ops](https://github.com/PhilRunninger/nerdtree-buffer-ops)
    - [ ] [PhilRunninger/nerdtree-visual-selection](https://github.com/PhilRunninger/nerdtree-visual-selection)
    - [ ] [jistr/vim-nerdtree-tabs](https://github.com/jistr/vim-nerdtree-tabs)
    - [ ] Others (specify):
- Provide a minimal **.vimrc** file that will reproduce the issue.
```vim
let g:NERDTreeMapPreview = 'o'
let g:NERDTreeWinSize = 30
let g:NERDTreeWinPos = 'right'
```

#### Steps to Reproduce the Issue
1.  Open NERDTree file tree
2. Press 'm' to open file menu
3. Press 'l' (or any other menu item)

#### Current Behavior (Include screenshots where appropriate.)

`cmdheight` is not restored and is left to size equal to menu size.

#### Expected Result

`cmdheight` should be restored back to previous value, in my case to `1`.

**Comment by sirex** [comment]
I found, that this might have something to do with these functions:

https://github.com/preservim/nerdtree/blob/81f3eaba295b3fceb2d032db57e5eae99ae480f8/lib/nerdtree/menu_controller.vim#L160-L175

**Comment by kaicoh** [comment]
same for me.

**Comment by fizyk20** [comment]
Having the same issue for some time now.

**Comment by jhang-jhe-wei** [comment]
I met the same problem, but after I upgrade the NeoVim version to `0.8.1`, the problem is fixed.

https://user-images.githubusercontent.com/59594475/207440600-91c5ecb9-f117-4cc9-81aa-27157a30f458.mp4

**Comment by nickrivadeneira** [comment]
Upgrading to 0.8.1 does seem to fix the issue for me as well. Thanks for sharing!

**Comment by jhang-jhe-wei** [comment]
https://github.com/neovim/neovim/releases/tag/v0.8.1
<img width="916" alt="image" src="https://user-images.githubusercontent.com/59594475/207441501-3bc0af47-3b03-492f-b8ff-19fec15c24e6.png">

**Comment by bokutin** [comment]
I was having the same problem.

The following changes fixed it for the moment.

<pre>
--- nerdtree/lib/nerdtree/menu_controller.vim.orig
+++ nerdtree/lib/nerdtree/menu_controller.vim
@@ -170,6 +170,7 @@
 "FUNCTION: MenuController._restoreOptions() {{{1
 "restore the options we saved in _saveOptions()
 function! s:MenuController._restoreOptions()
+    let &cmdheight = self._oldCmdheight + 1
     let &cmdheight = self._oldCmdheight
     let &lazyredraw = self._oldLazyredraw
 endfunction
</pre>

I am using editors/vim on FreeBSD 2022Q4 ports, but I wonder if it is a vim problem.

**Comment by grwlf** [comment]
> I was having the same problem.
> 
> The following changes fixed it for the moment.
> 
> --- nerdtree/lib/nerdtree/menu_controller.vim.orig
> +++ nerdtree/lib/nerdtree/menu_controller.vim
> @@ -170,6 +170,7 @@
>  "FUNCTION: MenuController._restoreOptions() {{{1
>  "restore the options we saved in _saveOptions()
>  function! s:MenuController._restoreOptions()
> +    let &cmdheight = self._oldCmdheight + 1
>      let &cmdheight = self._oldCmdheight
>      let &lazyredraw = self._oldLazyredraw
>  endfunction
> 
> I am using editors/vim on FreeBSD 2022Q4 ports, but I wonder if it is a vim problem.

Thanks! This fixes the issue in Vim (I am on vim-9.0)!
I noticed that sometimes after I add files using NERDTree menu, I get its window scrolled down to the bottom. Not sure if it is related or not.
